base/iobuffer.jl

```diff
@@ -260,10 +259,7 @@ function takebuf_array(io::IOBuffer)
     end
     return data
 end
-function takebuf_string(io::AbstractIOBuffer)
-    b = takebuf_array(io)
-    return isvalid(ASCIIString, b) ? ASCIIString(b) : UTF8String(b)
-end
+takebuf_string(io::AbstractIOBuffer) = String(takebuf_array(io))
```

Comment:
Looks like we'll be able to make the API simpler now by removing `takebuf_string` and renaming `takebuf_array` to something shorter.